internal/backend/kubernetes/application/kubernetes.go

```diff
@@ -82,7 +89,15 @@ func (be *KubernetesBackend) Create(ctx context.Context, app *v1alpha1.Applicati
 }
 
 func (be *KubernetesBackend) Get(ctx context.Context, name string, namespace string) (*v1alpha1.Application, error) {
-	return be.appClient.ArgoprojV1alpha1().Applications(namespace).Get(ctx, name, v1.GetOptions{})
+	obj, err := be.appLister.ByNamespace(namespace).Get(name)
+	if err != nil {
+		return nil, err
+	}
+	app, ok := obj.(*v1alpha1.Application)
+	if !ok {
+		return nil, fmt.Errorf("object is not an Application: %T", obj)
+	}
+	return app, nil
```

Comment:
Items returned from the cache need to be treated read-only. I believe that's why unrelated unit tests are failing and panicing.

There are two options imho:

1. We clearly document that objects returned by this function are to be treated read-only, because they are retrieved directly from the cache and caller needs to make a copy if they want to modify it, or
2. We return a copy of the object retrieved from the cache, to take the burden from the caller.

The first option puts more responsibility to the caller, but is resource efficient. The second option would ensure that the caller can treat the objects lightly, but for the cost of increased memory consumption.

I have not yet made up my mind with regards to which solution I'd prefer. Which one do you think makes more sense?